src/apispec/ext/marshmallow/field_converter.py

```diff
@@ -439,7 +439,16 @@ def nested2properties(self, field: marshmallow.fields.Field, ret) -> dict:
         ):
             schema_dict = self.resolve_nested_schema(field.schema)  # type:ignore
             if ret and "$ref" in schema_dict:
-                ret.update({"allOf": [schema_dict]})
+                if "type" in ret:
+                    type = ret.pop("type")
+                    if isinstance(type, list):
+                        types = list(map(lambda t: {"type": t}, type))
```

Comment:
using `map` in python... controversial 😆 - A list comprehension tends to be the preferred approach if you don't want the pythonistas on your back
```suggestion
                        types = [{"type": t} for t in type]
```

Reply:
as much as I love to rattle the pythonistas, addressed this one